doc-tool/src/dotty/tools/dottydoc/DocFrontEnd.scala

```diff
@@ -0,0 +1,17 @@
+package dotty.tools
+package dottydoc
+
+import dotc.typer.FrontEnd
+import dotc.core.Contexts.Context
+import dotc.CompilationUnit
+
+/** `DocFrontEnd` uses the Dotty `FrontEnd` without discarding the AnyVal
+  * interfaces for Boolean, Int, Char, Long, Byte etc.
+  *
+  * It currently still throws away Java sources by overriding
+  * `discardAfterTyper`.
```

Comment:
We should open an issue to support Java code like https://github.com/scala/scala/pull/5246